automation_tools/__init__.py

```diff
@@ -204,6 +205,9 @@ def setup_ddns(entry_domain, host_ip):
         fix_hostname(entry_domain, host_ip)
         run('redhat-ddns-client enable')
         run('redhat-ddns-client')
+    # unsubscribe and removed java as it will conflict satellite installation
+    unsubscribe()
```

Comment:
Try subscription and Catch un-subscription, so that any issues in between should not skip un-subscription.

Reply:
No thought process is that no exception handling is required here.

Reply:
Lets say u subscribed and then something goes wrong with lines below that like in `fix hostname` we just dont want to exit out of that function if that func is called with `warn_only=true`, coz that will skip unsubscription and hence failing later running tests.

Reply:
@jyejare This is the pre-setup part for running the test case. We should be halting the execution in case if the problem persists here. I wanted to have hard assert here rather than soft assert. If you have hard feelings that this needs to be added as part of try-catch I will add it.

Reply:
done !

Reply:
Just a Suggestion:
1. Catch the exception bu raise it.
2. Add finally block for unsubscribe so need not to call twice.

Example:
try:
    bla 
    bla
except:
    print('error!')
    raise Exception
finally:
    unsubscribe()

Reply:
you are right, added finally.